src/explore-education-statistics-admin/src/pages/release/data/components/__tests__/ApiDataSetCreateModal.test.tsx

```diff
@@ -37,19 +39,37 @@ describe('ApiDataSetCreateModal', () => {
         releaseId="release-id"
         onSubmit={noop}
       />,
+      { history },
     );
 
+    expect(await screen.findByText('Create API data set')).toBeInTheDocument();
+
     await user.click(
       await screen.findByRole('button', { name: 'Create API data set' }),
     );
 
+    expect(screen.getByRole('button', { name: 'Close' })).toBeInTheDocument();
+
     expect(
-      await screen.findByText(
+      screen.getByText(
         /No API data sets can be created as there are no candidate data files available/,
       ),
     ).toBeInTheDocument();
 
-    expect(screen.getByRole('button', { name: 'Close' })).toBeInTheDocument();
+    expect(
+      await screen.findByRole('link', { name: 'Data and files' }),
+    ).toBeInTheDocument();
+
+    await user.click(screen.getByRole('link', { name: 'Data and files' }));
+
+    expect(
+      screen.queryByRole('link', { name: 'Data and files' }),
+    ).not.toBeInTheDocument();
+
+    expect(history.location.pathname).toEqual(
+      '/publication/publication-id/release/release-id/data',
+    );
+    expect(history.location.hash).toEqual('#data-uploads');
```

Comment:
No need to click the link and navigate - just assert that the link has the correct `href`. Additionally, we can tighten up the checks to be within the modal.

We can replace these lines with the following:

```ts
const modal = within(screen.getByRole('dialog'));

expect(
  modal.getByText(
    /No API data sets can be created as there are no candidate data files available/,
  ),
).toBeInTheDocument();

expect(modal.getByRole('link', { name: 'Data and files' })).toHaveAttribute(
  'href',
  '/publication/publication-id/release/release-id/data#data-uploads',
);
```